### PR TITLE
Add config to destroy the k8s cluster after a specific time and manage k8s secrets in git

### LINKED ISSUE #24: Add config to destroy the k8s cluster after 4 hours since it started
### What's your idea?

Currently the k8s cluster once it is created runs ad infinitum. In order to control costs, we want it to be destroyed after a reasonable amount of time that it is needed for testing.

### Impact

High

### Urgency

Now

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #25: Commit k8s secret to git
### What's your idea?

Currently we cannot commit secrets to git as they would be exposed to the world. This forces us to keep them locally and prevents us to create k8s secrets in a cicd workflow. 
A solution to this is given by [Bitnami sealed-secrets](https://github.com/bitnami-labs/sealed-secrets#usage), which allows to encrypt secrets before committing them to git.

### Impact

High

### Urgency

Now

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by giulio-giunta**
## Description
This adds configuration to:

1. Destroy the cluster after a certain time from creation (at the moment 4 hours)
2. Encrypt k8s secrets so that they can be committed to git and used in a cicd workflow

## Related Issue
"resolves #24" and  "resolves #25"

## Motivation and Context

1. We need a way to destroy the cluster automatically after a certain amount of time when it is used for testing. 
2. We need to be able to commit secrets, so that git is the unique source of truth. This also allows us to use those secrets in cicd workflows when deploying k8s resources.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTING](https://github.com/GeekZoneHQ/web/blob/main/CONTRIBUTING.md)** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: aws/k8s/sealed-secrets/sealing-key.pem [hunk 1]
```diff
@@ -0,0 +1,28 @@
+-----BEGIN CERTIFICATE-----
```
Comment (jamesgeddes) [inline aws/k8s/sealed-secrets/sealing-key.pem]: Is this a public cert?

File: azure/k8s/sealed-secrets/master-key-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
...SmJ1cVVLK3NiRmN4dmlQa21qQkhqSVFwMXV1c1YySlQ5SXdqMmM5cDZVU1ZlNCsKY28rWXBRSURBUUFCbzBJd1FEQU9CZ05WSFE4QkFmOEVCQU1DQUFFd0R3WURWUjBUQVFIL0JBVXdBd0VCL3pBZApCZ05WSFE0RUZnUVVYVlhhZmJPK21xbm9qWndOYURpZHhWQjBIcmt3RFFZSktvWklodmNOQVFFTEJRQURnZ0lCCkFMSjZjSy9Uc3RtUVZSbkp2Y29HNUZQTGh4SXBqQkZheUFzc3c5MVdHdU9nTVhmVTJmQWNBcnNRY3Y3YzIvQUwKcVkwVFJlaklrdTFLQzlvcHRzdnV4QmFIMnEra3gydFFxSUltR0J0RlhZZmd0bVI3Mi8ybFNwZWUzTFZTODR2RQpOQmhhZHhvOEtRVmJiNkh0TE82Qml0ODFaZitYNnBxaWRNbklqeGFuUkJZZ2JGOVZ4Sm1EczlzenZtMDgwUjhyCkRtbWorU2NaalZ6cGtJaGpJeW5QNGU0UFA4YTd1dGR6bUNEcE5odWx3blNUN21SSk5mcitwV3FmRmRaaFBVSWwKbExlUWN3aEtmMllYbGpLeUtmUHNqWEhZblNKc1ZabVVwZ0ZUeWtzQVYzU0w5UzFIQWNZNUFWa0o1alc3MmU2QQprcVBUS1dESGZhU3BMOFNEQmh3UmhLNnhSN3l2ZHBmc0N1OFQzTmJaOG9lSWwzVk5HMitLTGNLQnlsRlZiSFFzCnhyWUFZSTRHa3Q1Mjk0cTdpTGo3OWNBN3RnL2N2NzJRY09TYlJGTTV5RFlweHNDOGVkdlZTRmFyd1hQNDNmSDMKTmNkaFpUK0FvQ0FQNHNJOHpJWks0SWpNaXRud2FremZsZHl6RU1OVm9sL3lwQlgrM01IRHcxOWdNSkdUNjNUTgp4S28zZW5FeWNoT25WU1c0OWwzaDlhQm9lMEhtRXQ3OVNZM0ZmYlVFM2hXVDBEZDQyb0xrTThuYTh6WU52R2JDClYxdXpZV2FGMk9qM1hmbmFqcTFZZytSdiswd1h0akZtZkxsUytQM09HYm50SERDdlpXYmtOYWVHUkFYcnlmL08KZUowQm1wMXBXMDgrQk56M21kZHZnS0dmMkdITjFZOThoNXcxTlNRbGw3azkKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
```
Comment (jamesgeddes) [inline azure/k8s/sealed-secrets/master-key-secret.yaml:3]: Are we sure that this is OK to publish?

File: azure/k8s/sealed-secrets/sealing-key.pem [hunk 3]
```diff
@@ -0,0 +1,28 @@
+-----BEGIN CERTIFICATE-----
```
Comment (jamesgeddes) [inline azure/k8s/sealed-secrets/sealing-key.pem]: Is this a public cert?

**Review by jamesgeddes — CHANGES_REQUESTED** [review]

File: aws/k8s/sealed-secrets/sealing-key.pem (hunk 1 above)
Comment (giulio-giunta) [reply aws/k8s/sealed-secrets/sealing-key.pem]: Yes, that's used to encrypt secrets. The private key is secret and used by a controller running in the cluster to decrypt sealed-secrets and create secrets out of them.

File: azure/k8s/sealed-secrets/master-key-secret.yaml (hunk 2 above)
Comment (giulio-giunta) [reply azure/k8s/sealed-secrets/master-key-secret.yaml:3]: Yes, it's a public key, as such cannot be used to decrypt secrets. It's not crucial to push to github, but can be used to encrypt secrets offline, without the cluster running.

File: azure/k8s/sealed-secrets/sealing-key.pem (hunk 3 above)
Comment (giulio-giunta) [reply azure/k8s/sealed-secrets/sealing-key.pem]: Yes

File: aws/k8s/sealed-secrets/master-key-secret.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  tls.key: ${TLS_KEY}
+kind: Secret
+metadata:
+  labels:
+    sealedsecrets.bitnami.com/sealed-secrets-key: active
+  name: sealed-secrets-key
+  namespace: kube-system
+type: kubernetes.io/tls
```
Comment (jamesgeddes) [inline aws/k8s/sealed-secrets/master-key-secret.yaml:11]: Does this file need to be called `master`*?

File: azure/k8s/sealed-secrets/master-key-secret.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  tls.key: ${TLS_KEY}
+kind: Secret
+metadata:
+  labels:
+    sealedsecrets.bitnami.com/sealed-secrets-key: active
+  name: sealed-secrets-key
+  namespace: kube-system
+type: kubernetes.io/tls
```
Comment (jamesgeddes) [inline azure/k8s/sealed-secrets/master-key-secret.yaml:11]: Does this file need to be called `master`*?

File: azure/create-azure-infra [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+cd azure/terraform-azure/
+terraform init
+terraform apply -auto-approve
+
+# Create resources in the k8s cluster
+cd ../k8s/
+az aks get-credentials --resource-group geekzone --name GeekZoneCluster --admin
+kubectl apply -f namespaces.yaml
+envsubst < sealed-secrets/master-key-secret.yaml | sponge sealed-secrets/master-key-secret.yaml
+kubectl apply -f sealed-secrets/master-key-secret.yaml
```
Comment (jamesgeddes) [inline azure/create-azure-infra]: Should be renamed to reflect inclusive naming

File: aws/create-aws-infra [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+terraform init
+terraform apply -auto-approve
+
+# Create resources in the k8s cluster
+cd ../k8s/
+rm ~/.kube/config
+aws eks --region eu-west-2 update-kubeconfig --name GeekZoneCluster 
+kubectl apply -f namespaces.yaml
+envsubst < sealed-secrets/master-key-secret.yaml | sponge sealed-secrets/master-key-secret.yaml
+kubectl apply -f sealed-secrets/master-key-secret.yaml
```
Comment (jamesgeddes) [inline aws/create-aws-infra]: Should be renamed to reflect inclusive naming

**Review by jamesgeddes — CHANGES_REQUESTED** [review]
Sorry @giulio-giunta - I should have picked up on the inclusive naming aspect in my previous review. It will form part of our CI in the future 😄

**Review by jamesgeddes — APPROVED** [review]
Discussed with @giulio-giunta and we will update the names in a separate ticket

**Pull request merged**